Bugs/2581 windows readkey hack

haf opened the pull request:
Equiv to https://github.com/fsharp/fsharp/pull/679

msftclas commented:
@__haf__,
Thanks for your contribution.
To ensure that the project team has proper rights to use your work, please complete the Contribution License Agreement at https://cla.microsoft.com.

It will cover your contributions to all Microsoft-managed open source projects.
_Thanks,
Microsoft Pull Request Bot_

msftclas commented:
__@haf__, thanks for signing the contribution license agreement. We will now validate the agreement and then the pull request.
<br />_Thanks, Microsoft Pull Request Bot_

dsyme commented on src/fsharp/fsi/console.fs:
```diff
@@ -17,7 +17,7 @@ module internal ConsoleOptions =
 #else
   // Bug 4254 was fixed in Dev11 (Net4.5), so this flag tracks making this fix up version specific.
-  let fixupRequired = not FSharpEnvironment.IsRunningOnNetFx45OrAbove
+  let fixupRequired = not FSharpEnvironment.IsRunningOnNetFx45OrAbove && Type.GetType("Mono.Runtime") = null
```
Use ``not runningOnMono`` ?

haf replied:
Sure.

I was hoping to get some feedback on the whole thing. Now that we've established it's a bug, is this a good fix? What about .net Core, for example? `IsRunningOnNetFx45OrAbove` is vacuously true both for mono and .Net Core, for example.

dsyme replied:
Does the fix work on your machine? What are you trying to achieve? That the fixup is used on Mono or not?   I don't think this code is used on .NET Core

haf replied:
As far as I can interpret the issues referenced, the problem with vscode garbling utf8 chars is because the fixup code runs; in error. I'm trying to make it not run. I haven't tried it on my machine, because I don't at this point know how to replace what interactive that vscode uses. Also, I'm on OS X, not ubuntu, but perhaps build.sh works there as well?

In general, I'm trying to nudge things in the right direction, because not having a REPL that supports UTF8 is generally an issue for me.

So; hypothesis; this fixes it. I don't know how to test it, but perhaps this repository has a README that documents it?

dsyme replied:
OK, I see. So I think the condition should never run when Mono is running, so just use this?

```
let fixupRequired = not FSharpEnvironment.IsRunningOnNetFx45OrAbove && not runningOnMono
```

But please verify by hand that this solves the problem.

dsyme commented on src/fsharp/fsi/console.fs:
```diff
@@ -46,56 +46,52 @@ type internal Style = Prompt | Out | Error
 type internal History() =
     let list  = new List<string>()
-    let mutable current  = 0 
+    let mutable current  = 0
```
Please remove all these whitespace changes, thanks.

KevinRansom replied:
@dsyme:  https://github.com/Microsoft/visualfsharp/pull/2582/files?w=1

haf commented:
I've updated the PR based on your feedback. However, I don't have the bandwidth to test this fix because this repository does not compile:

```
➜  visualfsharp git:(bugs/2581-windows-readkey-hack) ./build.sh
Build/Tests configuration:

BUILD_PROTO=0
BUILD_PROTO_WITH_CORECLR_LKG=0
BUILD_NET40=1
BUILD_CORECLR=0
BUILD_PORTABLE=0
BUILD_VS=0
BUILD_SETUP=
BUILD_CONFIG=release
BUILD_PUBLICSIGN=0

TEST_NET40_COMPILERUNIT_SUITE=0
TEST_NET40_COREUNIT_SUITE=0
TEST_NET40_FSHARP_SUITE=0
TEST_NET40_FSHARPQA_SUITE=0
TEST_CORECLR_COREUNIT_SUITE=0
TEST_CORECLR_FSHARP_SUITE=0
TEST_PORTABLE_COREUNIT_SUITE=0
TEST_VS_IDEUNIT_SUITE=0
INCLUDE_TEST_SPEC_NUNIT=
INCLUDE_TEST_TAGS=

Environment

(TODO)


---------------- Done with arguments, starting preparation -----------------
Restoring NuGet package System.ValueTuple.4.3.0.
Restoring NuGet package System.Collections.Immutable.1.2.0.
Restoring NuGet package Microsoft.DiaSymReader.PortablePdb.1.1.0.
Restoring NuGet package FsLexYacc.7.0.4.
Restoring NuGet package MicroBuild.Core.0.2.0.
Restoring NuGet package MicroBuild.Core.Sentinel.1.0.0.
Restoring NuGet package System.Reflection.Metadata.1.4.1-beta-24227-04.
Restoring NuGet package Microsoft.DiaSymReader.1.0.8.
Restoring NuGet package Microsoft.VisualFSharp.Msbuild.15.0.1.0.1.
Restoring NuGet package FsCheck.2.6.2.
Restoring NuGet package NUnit.3.5.0.
Adding package 'System.ValueTuple.4.3.0' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
Adding package 'System.Reflection.Metadata.1.4.1-beta-24227-04' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
Adding package 'System.Collections.Immutable.1.2.0' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
Restoring NuGet package NUnit.Console.3.0.0.
Restoring NuGet package NUnitLite.3.5.0.
Added package 'System.ValueTuple.4.3.0' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
Restoring NuGet package NUnit3TestAdapter.3.0.8-ctp-8.
Added package 'System.Collections.Immutable.1.2.0' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
Added package 'System.Reflection.Metadata.1.4.1-beta-24227-04' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
Restoring NuGet package BenchmarkDotNet.0.9.8.
Restoring NuGet package BenchmarkDotNet.Diagnostics.Windows.0.9.8.
Adding package 'FsCheck.2.6.2' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
Added package 'FsCheck.2.6.2' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
Restoring NuGet package VisualCppTools.14.0.24519-Pre.
Restoring NuGet package Newtonsoft.Json.6.0.8.
Restoring NuGet package Microsoft.FSharp.TupleSample.1.0.0-alpha-161121.
Restoring NuGet package Microsoft.VSSDK.BuildTools.15.0.26124-RC3.
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='FsLexYacc',Version='7.0.4')
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='MicroBuild.Core',Version='0.2.0')
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='NUnit3TestAdapter',Version='3.0.8-ctp-8')
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='MicroBuild.Core.Sentinel',Version='1.0.0')
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='Microsoft.DiaSymReader',Version='1.0.8')
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='Microsoft.VisualFSharp.Msbuild.15.0',Version='1.0.1')
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='NUnit',Version='3.5.0')
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='Microsoft.DiaSymReader.PortablePdb',Version='1.1.0')
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='NUnit.Console',Version='3.0.0')
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='NUnitLite',Version='3.5.0')
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='BenchmarkDotNet',Version='0.9.8')
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='VisualCppTools',Version='14.0.24519-Pre')
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='BenchmarkDotNet.Diagnostics.Windows',Version='0.9.8')
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='Newtonsoft.Json',Version='6.0.8')
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='Microsoft.FSharp.TupleSample',Version='1.0.0-alpha-161121')
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='Microsoft.VSSDK.BuildTools',Version='15.0.26124-RC3')
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/microbuild.core.sentinel/1.0.0/microbuild.core.sentinel.1.0.0.nupkg
Adding package 'Microsoft.DiaSymReader.1.0.8' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
Added package 'Microsoft.DiaSymReader.1.0.8' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
Restoring NuGet package Microsoft.VisualStudio.Shell.14.0.14.3.25407.
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/microsoft.visualstudio.shell.14.0/14.3.25407/microsoft.visualstudio.shell.14.0.14.3.25407.nupkg
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='Microsoft.VisualStudio.Shell.14.0',Version='14.3.25407')
Adding package 'Microsoft.DiaSymReader.PortablePdb.1.1.0' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
Added package 'Microsoft.DiaSymReader.PortablePdb.1.1.0' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
Restoring NuGet package Microsoft.VisualStudio.Shell.Immutable.10.0.10.0.30319.
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/microsoft.visualstudio.shell.immutable.10.0/10.0.30319/microsoft.visualstudio.shell.immutable.10.0.10.0.30319.nupkg
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='Microsoft.VisualStudio.Shell.Immutable.10.0',Version='10.0.30319')
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/microbuild.core/0.2.0/microbuild.core.0.2.0.nupkg
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='NUnit.Console',Version='3.0.0') 8426ms
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='Microsoft.VisualFSharp.Msbuild.15.0',Version='1.0.1') 8426ms
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='Microsoft.VisualFSharp.Msbuild.15.0'
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='NUnit.Console'
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='NUnitLite',Version='3.5.0') 8439ms
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='NUnitLite'
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='BenchmarkDotNet.Diagnostics.Windows',Version='0.9.8') 8440ms
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='BenchmarkDotNet.Diagnostics.Windows'
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='Microsoft.VSSDK.BuildTools',Version='15.0.26124-RC3') 8440ms
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='Microsoft.VSSDK.BuildTools'
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='MicroBuild.Core.Sentinel',Version='1.0.0') 8455ms
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='MicroBuild.Core.Sentinel'
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='Microsoft.FSharp.TupleSample',Version='1.0.0-alpha-161121') 8443ms
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='Microsoft.FSharp.TupleSample'
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='Newtonsoft.Json',Version='6.0.8') 8452ms
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='MicroBuild.Core',Version='0.2.0') 8466ms
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='Newtonsoft.Json'
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='MicroBuild.Core'
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='BenchmarkDotNet',Version='0.9.8') 8453ms
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='FsLexYacc',Version='7.0.4') 8466ms
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='BenchmarkDotNet'
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='NUnit',Version='3.5.0') 8466ms
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='FsLexYacc'
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='VisualCppTools',Version='14.0.24519-Pre') 8452ms
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='NUnit'
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='VisualCppTools'
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='NUnit3TestAdapter',Version='3.0.8-ctp-8') 8467ms
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='NUnit3TestAdapter'
  NotFound https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/microbuild.core.sentinel/1.0.0/microbuild.core.sentinel.1.0.0.nupkg 2117ms
  NotFound https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/microsoft.visualstudio.shell.14.0/14.3.25407/microsoft.visualstudio.shell.14.0.14.3.25407.nupkg 1990ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='Microsoft.VisualFSharp.Msbuild.15.0' 176ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='NUnit.Console' 176ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='NUnitLite' 171ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='Microsoft.VSSDK.BuildTools' 170ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='BenchmarkDotNet.Diagnostics.Windows' 177ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='MicroBuild.Core.Sentinel' 178ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='NUnit' 178ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='Microsoft.FSharp.TupleSample' 186ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='Newtonsoft.Json' 184ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='MicroBuild.Core' 198ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='NUnit3TestAdapter' 197ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='BenchmarkDotNet' 198ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='FsLexYacc' 202ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='VisualCppTools' 209ms
  NotFound https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/microsoft.visualstudio.shell.immutable.10.0/10.0.30319/microsoft.visualstudio.shell.immutable.10.0.10.0.30319.nupkg 1850ms
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='Microsoft.VisualStudio.Shell.Immutable.10.0',Version='10.0.30319') 1909ms
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='Microsoft.VisualStudio.Shell.Immutable.10.0'
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='Microsoft.VisualStudio.Shell.14.0',Version='14.3.25407') 2177ms
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='Microsoft.VisualStudio.Shell.14.0'
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/microbuild.core/0.2.0/microbuild.core.0.2.0.nupkg
  NotFound https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/microbuild.core/0.2.0/microbuild.core.0.2.0.nupkg 1984ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='Microsoft.VisualStudio.Shell.Immutable.10.0' 168ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='Microsoft.VisualStudio.Shell.14.0' 169ms
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microbuild.core/0.2.0/microbuild.core.0.2.0.nupkg
  NotFound https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/microbuild.core/0.2.0/microbuild.core.0.2.0.nupkg 808ms
  OK https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microbuild.core/0.2.0/microbuild.core.0.2.0.nupkg 704ms
Acquiring lock for the installation of MicroBuild.Core 0.2.0
Acquired lock for the installation of MicroBuild.Core 0.2.0
Installing MicroBuild.Core 0.2.0.
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/microbuild.core/0.2.0/microbuild.core.0.2.0.nupkg
Completed installation of MicroBuild.Core 0.2.0
Adding package 'MicroBuild.Core.0.2.0' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/fslexyacc/7.0.4/fslexyacc.7.0.4.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/fslexyacc/7.0.4/fslexyacc.7.0.4.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/fslexyacc/7.0.4/fslexyacc.7.0.4.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/fslexyacc/7.0.4/fslexyacc.7.0.4.nupkg
Added package 'MicroBuild.Core.0.2.0' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
Restoring NuGet package FSharp.Compiler.Tools.4.0.1.21.
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/fsharp.compiler.tools/4.0.1.21/fsharp.compiler.tools.4.0.1.21.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/fsharp.compiler.tools/4.0.1.21/fsharp.compiler.tools.4.0.1.21.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/fsharp.compiler.tools/4.0.1.21/fsharp.compiler.tools.4.0.1.21.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/fsharp.compiler.tools/4.0.1.21/fsharp.compiler.tools.4.0.1.21.nupkg
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='FSharp.Compiler.Tools',Version='4.0.1.21')
  NotFound https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/fslexyacc/7.0.4/fslexyacc.7.0.4.nupkg 114ms
  NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/fslexyacc/7.0.4/fslexyacc.7.0.4.nupkg 169ms
  NotFound https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/fslexyacc/7.0.4/fslexyacc.7.0.4.nupkg 170ms
  NotFound https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/fsharp.compiler.tools/4.0.1.21/fsharp.compiler.tools.4.0.1.21.nupkg 146ms
  NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/fsharp.compiler.tools/4.0.1.21/fsharp.compiler.tools.4.0.1.21.nupkg 251ms
  NotFound https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/fsharp.compiler.tools/4.0.1.21/fsharp.compiler.tools.4.0.1.21.nupkg 252ms
  NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/fslexyacc/7.0.4/fslexyacc.7.0.4.nupkg 690ms
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/fslexyacc/7.0.4/fslexyacc.7.0.4.nupkg
  NotFound https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='FSharp.Compiler.Tools',Version='4.0.1.21') 670ms
  GET https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='FSharp.Compiler.Tools'
  NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/fsharp.compiler.tools/4.0.1.21/fsharp.compiler.tools.4.0.1.21.nupkg 772ms
  OK https://vcppdogfooding.azurewebsites.net/nuget/FindPackagesById()?id='FSharp.Compiler.Tools' 168ms
  GET https://api.nuget.org/v3-flatcontainer/fslexyacc/7.0.4/fslexyacc.7.0.4.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/fslexyacc/7.0.4/fslexyacc.7.0.4.nupkg
  OK https://api.nuget.org/v3-flatcontainer/fslexyacc/7.0.4/fslexyacc.7.0.4.nupkg 451ms
Acquiring lock for the installation of FsLexYacc 7.0.4
Acquired lock for the installation of FsLexYacc 7.0.4
Installing FsLexYacc 7.0.4.
  NotFound https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/fslexyacc/7.0.4/fslexyacc.7.0.4.nupkg 671ms
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/fslexyacc/7.0.4/fslexyacc.7.0.4.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/microbuild.core.sentinel/1.0.0/microbuild.core.sentinel.1.0.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microbuild.core.sentinel/1.0.0/microbuild.core.sentinel.1.0.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/microbuild.core.sentinel/1.0.0/microbuild.core.sentinel.1.0.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/microbuild.core.sentinel/1.0.0/microbuild.core.sentinel.1.0.0.nupkg
  GET https://api.nuget.org/v3-flatcontainer/microbuild.core.sentinel/1.0.0/microbuild.core.sentinel.1.0.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/microbuild.core.sentinel/1.0.0/microbuild.core.sentinel.1.0.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/microbuild.core.sentinel/1.0.0/microbuild.core.sentinel.1.0.0.nupkg
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/microsoft.visualfsharp.msbuild.15.0/1.0.1/microsoft.visualfsharp.msbuild.15.0.1.0.1.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/microsoft.visualfsharp.msbuild.15.0/1.0.1/microsoft.visualfsharp.msbuild.15.0.1.0.1.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microsoft.visualfsharp.msbuild.15.0/1.0.1/microsoft.visualfsharp.msbuild.15.0.1.0.1.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/microsoft.visualfsharp.msbuild.15.0/1.0.1/microsoft.visualfsharp.msbuild.15.0.1.0.1.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/microsoft.visualfsharp.msbuild.15.0/1.0.1/microsoft.visualfsharp.msbuild.15.0.1.0.1.nupkg
  GET https://api.nuget.org/v3-flatcontainer/microsoft.visualfsharp.msbuild.15.0/1.0.1/microsoft.visualfsharp.msbuild.15.0.1.0.1.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/microsoft.visualfsharp.msbuild.15.0/1.0.1/microsoft.visualfsharp.msbuild.15.0.1.0.1.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/microsoft.visualfsharp.msbuild.15.0/1.0.1/microsoft.visualfsharp.msbuild.15.0.1.0.1.nupkg
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/nunit/3.5.0/nunit.3.5.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/nunit/3.5.0/nunit.3.5.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/nunit/3.5.0/nunit.3.5.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/nunit/3.5.0/nunit.3.5.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/nunit/3.5.0/nunit.3.5.0.nupkg
  GET https://api.nuget.org/v3-flatcontainer/nunit/3.5.0/nunit.3.5.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/nunit/3.5.0/nunit.3.5.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/nunit/3.5.0/nunit.3.5.0.nupkg
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/nunit.console/3.0.0/nunit.console.3.0.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/nunit.console/3.0.0/nunit.console.3.0.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/nunit.console/3.0.0/nunit.console.3.0.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/nunit.console/3.0.0/nunit.console.3.0.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/nunit.console/3.0.0/nunit.console.3.0.0.nupkg
  GET https://api.nuget.org/v3-flatcontainer/nunit.console/3.0.0/nunit.console.3.0.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/nunit.console/3.0.0/nunit.console.3.0.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/nunit.console/3.0.0/nunit.console.3.0.0.nupkg
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/nunit3testadapter/3.0.8-ctp-8/nunit3testadapter.3.0.8-ctp-8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/nunit3testadapter/3.0.8-ctp-8/nunit3testadapter.3.0.8-ctp-8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/nunit3testadapter/3.0.8-ctp-8/nunit3testadapter.3.0.8-ctp-8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/nunit3testadapter/3.0.8-ctp-8/nunit3testadapter.3.0.8-ctp-8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/nunit3testadapter/3.0.8-ctp-8/nunit3testadapter.3.0.8-ctp-8.nupkg
  GET https://api.nuget.org/v3-flatcontainer/nunit3testadapter/3.0.8-ctp-8/nunit3testadapter.3.0.8-ctp-8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/nunit3testadapter/3.0.8-ctp-8/nunit3testadapter.3.0.8-ctp-8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/nunit3testadapter/3.0.8-ctp-8/nunit3testadapter.3.0.8-ctp-8.nupkg
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/nunitlite/3.5.0/nunitlite.3.5.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/nunitlite/3.5.0/nunitlite.3.5.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/nunitlite/3.5.0/nunitlite.3.5.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/nunitlite/3.5.0/nunitlite.3.5.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/nunitlite/3.5.0/nunitlite.3.5.0.nupkg
  GET https://api.nuget.org/v3-flatcontainer/nunitlite/3.5.0/nunitlite.3.5.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/nunitlite/3.5.0/nunitlite.3.5.0.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/nunitlite/3.5.0/nunitlite.3.5.0.nupkg
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/benchmarkdotnet/0.9.8/benchmarkdotnet.0.9.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/benchmarkdotnet/0.9.8/benchmarkdotnet.0.9.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/benchmarkdotnet/0.9.8/benchmarkdotnet.0.9.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/benchmarkdotnet/0.9.8/benchmarkdotnet.0.9.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/benchmarkdotnet/0.9.8/benchmarkdotnet.0.9.8.nupkg
  GET https://api.nuget.org/v3-flatcontainer/benchmarkdotnet/0.9.8/benchmarkdotnet.0.9.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/benchmarkdotnet/0.9.8/benchmarkdotnet.0.9.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/benchmarkdotnet/0.9.8/benchmarkdotnet.0.9.8.nupkg
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/visualcpptools/14.0.24519-pre/visualcpptools.14.0.24519-pre.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/visualcpptools/14.0.24519-pre/visualcpptools.14.0.24519-pre.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/visualcpptools/14.0.24519-pre/visualcpptools.14.0.24519-pre.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/visualcpptools/14.0.24519-pre/visualcpptools.14.0.24519-pre.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/visualcpptools/14.0.24519-pre/visualcpptools.14.0.24519-pre.nupkg
  GET https://api.nuget.org/v3-flatcontainer/visualcpptools/14.0.24519-pre/visualcpptools.14.0.24519-pre.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/visualcpptools/14.0.24519-pre/visualcpptools.14.0.24519-pre.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/visualcpptools/14.0.24519-pre/visualcpptools.14.0.24519-pre.nupkg
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/benchmarkdotnet.diagnostics.windows/0.9.8/benchmarkdotnet.diagnostics.windows.0.9.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/benchmarkdotnet.diagnostics.windows/0.9.8/benchmarkdotnet.diagnostics.windows.0.9.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/benchmarkdotnet.diagnostics.windows/0.9.8/benchmarkdotnet.diagnostics.windows.0.9.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/benchmarkdotnet.diagnostics.windows/0.9.8/benchmarkdotnet.diagnostics.windows.0.9.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/benchmarkdotnet.diagnostics.windows/0.9.8/benchmarkdotnet.diagnostics.windows.0.9.8.nupkg
  GET https://api.nuget.org/v3-flatcontainer/benchmarkdotnet.diagnostics.windows/0.9.8/benchmarkdotnet.diagnostics.windows.0.9.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/benchmarkdotnet.diagnostics.windows/0.9.8/benchmarkdotnet.diagnostics.windows.0.9.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/benchmarkdotnet.diagnostics.windows/0.9.8/benchmarkdotnet.diagnostics.windows.0.9.8.nupkg
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/newtonsoft.json/6.0.8/newtonsoft.json.6.0.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/newtonsoft.json/6.0.8/newtonsoft.json.6.0.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/newtonsoft.json/6.0.8/newtonsoft.json.6.0.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/newtonsoft.json/6.0.8/newtonsoft.json.6.0.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/newtonsoft.json/6.0.8/newtonsoft.json.6.0.8.nupkg
  GET https://api.nuget.org/v3-flatcontainer/newtonsoft.json/6.0.8/newtonsoft.json.6.0.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/newtonsoft.json/6.0.8/newtonsoft.json.6.0.8.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/newtonsoft.json/6.0.8/newtonsoft.json.6.0.8.nupkg
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/microsoft.fsharp.tuplesample/1.0.0-alpha-161121/microsoft.fsharp.tuplesample.1.0.0-alpha-161121.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/microsoft.fsharp.tuplesample/1.0.0-alpha-161121/microsoft.fsharp.tuplesample.1.0.0-alpha-161121.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microsoft.fsharp.tuplesample/1.0.0-alpha-161121/microsoft.fsharp.tuplesample.1.0.0-alpha-161121.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/microsoft.fsharp.tuplesample/1.0.0-alpha-161121/microsoft.fsharp.tuplesample.1.0.0-alpha-161121.nupkg
  GET https://api.nuget.org/v3-flatcontainer/microsoft.fsharp.tuplesample/1.0.0-alpha-161121/microsoft.fsharp.tuplesample.1.0.0-alpha-161121.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/microsoft.fsharp.tuplesample/1.0.0-alpha-161121/microsoft.fsharp.tuplesample.1.0.0-alpha-161121.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/microsoft.fsharp.tuplesample/1.0.0-alpha-161121/microsoft.fsharp.tuplesample.1.0.0-alpha-161121.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/microsoft.fsharp.tuplesample/1.0.0-alpha-161121/microsoft.fsharp.tuplesample.1.0.0-alpha-161121.nupkg
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/microsoft.vssdk.buildtools/15.0.26124-rc3/microsoft.vssdk.buildtools.15.0.26124-rc3.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/microsoft.vssdk.buildtools/15.0.26124-rc3/microsoft.vssdk.buildtools.15.0.26124-rc3.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microsoft.vssdk.buildtools/15.0.26124-rc3/microsoft.vssdk.buildtools.15.0.26124-rc3.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/microsoft.vssdk.buildtools/15.0.26124-rc3/microsoft.vssdk.buildtools.15.0.26124-rc3.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/microsoft.vssdk.buildtools/15.0.26124-rc3/microsoft.vssdk.buildtools.15.0.26124-rc3.nupkg
  GET https://api.nuget.org/v3-flatcontainer/microsoft.vssdk.buildtools/15.0.26124-rc3/microsoft.vssdk.buildtools.15.0.26124-rc3.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/microsoft.vssdk.buildtools/15.0.26124-rc3/microsoft.vssdk.buildtools.15.0.26124-rc3.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/microsoft.vssdk.buildtools/15.0.26124-rc3/microsoft.vssdk.buildtools.15.0.26124-rc3.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/microsoft.visualstudio.shell.14.0/14.3.25407/microsoft.visualstudio.shell.14.0.14.3.25407.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microsoft.visualstudio.shell.14.0/14.3.25407/microsoft.visualstudio.shell.14.0.14.3.25407.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/microsoft.visualstudio.shell.14.0/14.3.25407/microsoft.visualstudio.shell.14.0.14.3.25407.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/microsoft.visualstudio.shell.14.0/14.3.25407/microsoft.visualstudio.shell.14.0.14.3.25407.nupkg
  GET https://api.nuget.org/v3-flatcontainer/microsoft.visualstudio.shell.14.0/14.3.25407/microsoft.visualstudio.shell.14.0.14.3.25407.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/microsoft.visualstudio.shell.14.0/14.3.25407/microsoft.visualstudio.shell.14.0.14.3.25407.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/microsoft.visualstudio.shell.14.0/14.3.25407/microsoft.visualstudio.shell.14.0.14.3.25407.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/microsoft.visualstudio.shell.immutable.10.0/10.0.30319/microsoft.visualstudio.shell.immutable.10.0.10.0.30319.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microsoft.visualstudio.shell.immutable.10.0/10.0.30319/microsoft.visualstudio.shell.immutable.10.0.10.0.30319.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/microsoft.visualstudio.shell.immutable.10.0/10.0.30319/microsoft.visualstudio.shell.immutable.10.0.10.0.30319.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/microsoft.visualstudio.shell.immutable.10.0/10.0.30319/microsoft.visualstudio.shell.immutable.10.0.10.0.30319.nupkg
  GET https://api.nuget.org/v3-flatcontainer/microsoft.visualstudio.shell.immutable.10.0/10.0.30319/microsoft.visualstudio.shell.immutable.10.0.10.0.30319.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/microsoft.visualstudio.shell.immutable.10.0/10.0.30319/microsoft.visualstudio.shell.immutable.10.0.10.0.30319.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/microsoft.visualstudio.shell.immutable.10.0/10.0.30319/microsoft.visualstudio.shell.immutable.10.0.10.0.30319.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/fsharp.compiler.tools/4.0.1.21/fsharp.compiler.tools.4.0.1.21.nupkg
  GET https://api.nuget.org/v3-flatcontainer/fsharp.compiler.tools/4.0.1.21/fsharp.compiler.tools.4.0.1.21.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/fsharp.compiler.tools/4.0.1.21/fsharp.compiler.tools.4.0.1.21.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/fsharp.compiler.tools/4.0.1.21/fsharp.compiler.tools.4.0.1.21.nupkg
  NotFound https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/microsoft.visualfsharp.msbuild.15.0/1.0.1/microsoft.visualfsharp.msbuild.15.0.1.0.1.nupkg 106ms
Completed installation of FsLexYacc 7.0.4
  NotFound https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/fslexyacc/7.0.4/fslexyacc.7.0.4.nupkg 670ms
Adding package 'FsLexYacc.7.0.4' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
  NotFound https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/microbuild.core.sentinel/1.0.0/microbuild.core.sentinel.1.0.0.nupkg 168ms
  NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/microbuild.core.sentinel/1.0.0/microbuild.core.sentinel.1.0.0.nupkg 169ms
  NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microbuild.core.sentinel/1.0.0/microbuild.core.sentinel.1.0.0.nupkg 172ms
Added package 'FsLexYacc.7.0.4' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
Restoring NuGet package FSharp.Data.2.2.5.
  NotFound https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/nunit/3.5.0/nunit.3.5.0.nupkg 239ms
  OK https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/microbuild.core.sentinel/1.0.0/microbuild.core.sentinel.1.0.0.nupkg 272ms
Acquiring lock for the installation of MicroBuild.Core.Sentinel 1.0.0
Acquired lock for the installation of MicroBuild.Core.Sentinel 1.0.0
Installing MicroBuild.Core.Sentinel 1.0.0.
Completed installation of MicroBuild.Core.Sentinel 1.0.0
Adding package 'MicroBuild.Core.Sentinel.1.0.0' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
Added package 'MicroBuild.Core.Sentinel.1.0.0' to folder '/Users/h/dev/logibit/_scratch/visualfsharp/packages'
  NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/microsoft.visualfsharp.msbuild.15.0/1.0.1/microsoft.visualfsharp.msbuild.15.0.1.0.1.nupkg 337ms
  GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/fsharp.data/2.2.5/fsharp.data.2.2.5.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-master-nightly/nuget/v3/flatcontainer/fsharp.data/2.2.5/fsharp.data.2.2.5.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/fsharp.data/2.2.5/fsharp.data.2.2.5.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/fsharp.data/2.2.5/fsharp.data.2.2.5.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/fsharp.data/2.2.5/fsharp.data.2.2.5.nupkg
  GET https://api.nuget.org/v3-flatcontainer/fsharp.data/2.2.5/fsharp.data.2.2.5.nupkg
  GET https://vcppdogfooding.azurewebsites.net/nuget/Packages(Id='FSharp.Data',Version='2.2.5')
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn-tools/nuget/v3/flatcontainer/fsharp.data/2.2.5/fsharp.data.2.2.5.nupkg
  GET https://dotnetmyget.blob.core.windows.net/artifacts/roslyn/nuget/v3/flatcontainer/fsharp.data/2.2.5/fsharp.data.2.2.5.nupkg
  NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microsoft.visualfsharp.msbuild.15.0/1.0.1/microsoft.visualfsharp.msbuild.15.0.1.0.1.nupkg 349ms
An error was encountered when fetching 'GET https://api.nuget.org/v3-flatcontainer/microsoft.visualfsharp.msbuild.15.0/1.0.1/microsoft.visualfsharp.msbuild.15.0.1.0.1.nupkg'. The request will now be retried.
An error occurred while sending the request
  Error: SecureChannelFailure (The authentication or decryption has failed.)
  The authentication or decryption has failed.
  Error while sending TLS Alert (Fatal:InternalError): System.IO.IOException: Unable to write data to the transport connection: Cannot access a disposed object.
  Object name: 'System.Net.Sockets.Socket'.. ---> System.ObjectDisposedException: Cannot access a disposed object.
  Object name: 'System.Net.Sockets.Socket'.
    at System.Net.Sockets.Socket.ThrowIfDisposedAndClosed () [0x00021] in <f8255d9ef0594d18ae2c0d97286b9a80>:0
    at System.Net.Sockets.Socket.BeginSend (System.Byte[] buffer, System.Int32 offset, System.Int32 size, System.Net.Sockets.SocketFlags socket_flags, System.AsyncCallback callback, System.Object state) [0x00000] in <f8255d9ef0594d18ae2c0d97286b9a80>:0
    at System.Net.Sockets.NetworkStream.BeginWrite (System.Byte[] buffer, System.Int32 offset, System.Int32 size, System.AsyncCallback callback, System.Object state) [0x000b4] in <f8255d9ef0594d18ae2c0d97286b9a80>:0
     --- End of inner exception stack trace ---
    at System.Net.Sockets.NetworkStream.BeginWrite (System.Byte[] buffer, System.Int32 offset, System.Int32 size, System.AsyncCallback callback, System.Object state) [0x00113] in <f8255d9ef0594d18ae2c0d97286b9a80>:0
    at Mono.Security.Protocol.Tls.RecordProtocol.BeginSendRecord (Mono.Security.Protocol.Tls.ContentType contentType, System.Byte[] recordData, System.AsyncCallback callback, System.Object state) [0x00026] in <59bcb3a41e31492f91d9513566444e88>:0
    at Mono.Security.Protocol.Tls.RecordProtocol.BeginSendRecord (Mono.Security.Protocol.Tls.Handshake.HandshakeType handshakeType, System.AsyncCallback callback, System.Object state) [0x00024] in <59bcb3a41e31492f91d9513566444e88>:0
    at Mono.Security.Protocol.Tls.SslClientStream.BeginNegotiateHandshake (System.AsyncCallback callback, System.Object state) [0x0004c] in <59bcb3a41e31492f91d9513566444e88>:0
    at Mono.Security.Protocol.Tls.SslStreamBase.BeginNegotiateHandshake (Mono.Security.Protocol.Tls.SslStreamBase+InternalAsyncResult asyncResult) [0x00021] in <59bcb3a41e31492f91d9513566444e88>:0
```

dsyme commented:
@jack-pappas I know you've been compiling this on OSX - any chance you could you take a look at this?

@haf Let's try the fix on Linux too

rmunn commented:
I initially couldn't build this because the `runningOnMono` name was not found (since it's in the `Microsoft.FSharp.Compiler.AbstractIL.IL` namespace, which is not imported into `fsi/console.fs` at the moment). I fully-qualifying the name as `Microsoft.FSharp.Compiler.AbstractIL.IL.runningOnMono` and that got the code to build for me, but doing `open Microsoft.FSharp.Compiler.AbstractIL.IL` would probably have worked too.

rmunn commented:
Probably related to my `FSharp.Core.resources.dll` problem: https://bugzilla.xamarin.com/show_bug.cgi?id=52738 and http://stackoverflow.com/questions/42428907/f-xamarin-forms-dll-loading-error-after-upgrading-fsharp-core-nuget-package

rmunn commented on src/fsharp/fsi/console.fs:
```diff
@@ -17,7 +17,7 @@ module internal ConsoleOptions =
 #else
   // Bug 4254 was fixed in Dev11 (Net4.5), so this flag tracks making this fix up version specific.
-  let fixupRequired = not FSharpEnvironment.IsRunningOnNetFx45OrAbove
+  let fixupRequired = not FSharpEnvironment.IsRunningOnNetFx45OrAbove && not runningOnMono
```
What about using `System.Console.InputEncoding.IsSingleByte` here? (I.e., `let fixupRequired = not FSharpEnvironment.IsRunningOnNetFx45OrAbove && System.Console.InputEncoding.IsSingleByte`). Because I could see this failing on .Net Core, depending on what version it reports for the `System` DLL. https://docs.microsoft.com/en-us/dotnet/articles/core/versions/ appears to promise that the `System` DLL will report a version higher than 4.5, so in theory there's no danger of `IsRunningOnNetFx45OrAbove` becoming false on .Net Core. But I haven't yet been able to test that theory.

And using `System.Console.InputEncoding.IsSingleByte` would specifically target the buggy environments where the `readKeyFixup` function was actually necessary: console encodings set to single-byte (i.e., non-Unicode) code pages. That way, someone on Windows 7 with .Net 4.0, but who ran `chcp 65001` before running `fsi.exe`, would get the benefit of this bugfix, rather than having his UTF-8 console fail him. (With just the `not runningOnMono` check, our theoretical `chcp 65001` user would still suffer this bug).

haf commented:
There.

jack-pappas commented:
@dsyme Sure, I can take a look / give this a try on macOS with Mono.

KevinRansom commented:
@jack-pappas @haf ,

This looks like an okay change to me ... has this fixed things on mono or not?

Thanks

Kevin

dsyme commented:
I'll close this and reverse-integrate the recent fixes from fsharp/fsharp
